In [ ]:
# {"CompanyName": "youtube video id"}

existing_videos = {
    "GarmaGuard": "bGg9F_3KgHo",
    "Roadie": "y4Dqgk5CqD4",
    "TouchUp Cup": "B0mfuIbkxFU",
    "Deux": "P3TkGk32-5Y",
    "Fish Fixe": "AoZdeYKGuR4",
    "Hidrent": "AAAzdm3wYlI",
    "Dude Wipes": "HVVA6J037wM",
    "Love & Pebble": "9PBZy9j3H3I",
    "Pink Picasso": "30-fgqbOwPU",
    "Sheets Laundry Club": "Kqa7cGTqmb8",
    "Zach & Zoe Sweet Bee Farm": "ZVVvJjwzl6c",
    "Elf Grams": "vhgDo-stGq8",
    "Ornament Anchor": "FaiTeA_qRrQ",
    "Santa's Enchanted Mailbox": "574MZ9iJHBM",
    "Wendy's Gnome Shop": "twvDm_8SyqQ",
    "Banana Loca": "KcloX3-vGJQ",
    "Liberate": "cpxiIVgqgGA",
    "MAXPRO SmartConnect": "RPoAzg8XqMo",
    "Tenikle": "sXkmrD_oTVA",
    "Black Sands Entertainment": "NAe24u3wmyw",
    "Candi": "LLCTeQnCR0s",
    "Snactiv": "UJRO3CveD4k",
    "The SMART Tire Company": "C0IoF9LH7Ao",
    "Headlightz": "HcaGrhftNe4",
    "VaBroom": "ix83s8bImWg",
    "TA3": "iHJ9wDeQ7JQ",
    "Tania Speaks Organic Skincare": "9CLdf-GNHI8",
    "Tristen Ikaika": "ehgyV-g1br4",
    "Calm Strips": "IJpOvKFpdX8",
    "Kettle Gryp": "Wy0z2ravpCQ",
    "Ade + Ayo": "OTZ6eBwp4xc",
    "Behave Bras": "CG6HmTcPjvE",
    "Curie": "UWH0pikR5Tw",
    "NOGGIN Boss": "tKX3D1rFDRw",
    "Chapul": "otl7LBnjXSk",
    "Do Amore": "dIOjQ5SlSiA",
    "ootBox": "2qYwZZj5WNs",
    "SUNFLOW": "rtcWMsw9HwA",
    "BootayBag": "h_Zw7dHyES0",
    "GoOats": "eRlp5hKOQZ4",
    "Pooch Paper": "MEhJ8alv6_c",
    "Chill-N-Reel": "ciD1XwhJWXk",
    "Drifties": "et9jM8fezNw",
    "Springer": "2dimrH8lZGI",
    "Stryx": "0alIIxlxQ3Y",
    "Kent": "qfPGtbGYIfg",
    "Oogiebear": "9-xtuDb1-k4",
    "Banana Phone": "Zfv8FqzNoyQ",
    "Stealth Bros & Co.": "EGhVrPiH4W0",
    "The Woobles": "X-4rzLhaAUI",
    "Turbo Trusser": "S51PnK0_oqM",
    "Create A Castle": "WIcEFgXDVsY",
    "Mama O's Premium Kimchi": "xWhpX-3Q6x8",
    "Pumpkin Glove Scraper": "gCx3jP8KHvc",
    "Stakt": "U8GBcS7rXKM",
    "Storage Scholars": "iAmoMRvbDtE",
    "Woosh": "dtH-c4GcGUA",
    "Bridal Babes": "sn0ZJXBYz-Q",
    "Plufl": "TQw9ISabbhU",
    "Collars & Co.": "AeT8y2MrAJQ",
    "SquareOne": "Q0EiP96bPSY",
    "Boarderie": "9m9SsHuAlCI",
    "Ride FRSH": "2PWait_Hfvw",
    "HummViewer": "JrEOZhrz2GI",
    "ShredSkinz": "7FoDIan0tUQ",
    "ChessUp": "6Sp7nYr4tVE",
    "ZipString": "o8BDGeIaKeI",
    "Prime 6": "0HvCpejDDEg",
    "Eat Your Flowers": "9xMs3epjI3A",
    "Foam Party Hats": "-PDGTiKP_wc",
    "Flated": "OzDA3GylmmI",
    "Pluie": "NIishreO_Ek",
    "Sweetkiwi": "jjQcncsA4Us",
    "Woof": "c8HZ-7qED4U",
    "Autio": "0Bvp8wRY5Xc",
    "Chubby Buttons": "9_6fXlOSNDQ",
    "Crispy Cones": "ljEdoe3dF8g",
    "Tngnt Ski Bikes": "VgwPzNHX3gY",
    "Bleni Blends": "Ezp8mtRFfDw",
    "Happi Floss": "l9U3v3EUaXo",
    "LavaBox Portable Campfires": "yDwZvNq2dlQ",
    "The Ice Cream Canteen": "8skWDuP8ls0",
    "Cincha Travel Belt": "KRZvjjaFQ4k",
    "Parting Stone": "GqpSxECpxGw",
    "Dapper Boi": "mHBxkjiLL7I",
    "Honey Bunchies": "hjGrWKyIDg8",
    "Play Maysie": "kfRAYiy0WV0",
    "Tia Lupita": "Onc5ZHoNt6Q",
    "Nature's Wild Berry": "UOYm9XI6kYo",
    "Noshi": "2q7fdwH3yB0",
    "You Go Natural": "eXX7nJfK6gg",
    "Krapp Strapp": "8S4UvJKuoDE",
    "Mini Materials": "TNdHQXMBCr4",
    "Supermix Studio": "9-magDe2vn4",
    "The Fidget Game": "pUT8bhtw91U",
    "Bunny Eyez": "Ce0PqzfB_Gc",
    "TIK PIK": "lRRu7iD8gkg",
    'SPERGO': 'tXYV-b4O1a8',
    '54 Thrones': 'J4qidvhYjvw',
    'SoaPen': 'sQO6BBrTNQ0',
    'Tabby': '7FaQ0uR1Fak',
    'TheMagic5': '27RPQNdb0iA',
    'Super Potty Trainer': 'OS8Y96pCIuA',
    'LIT Handlers': 'JZ3kezqk__Q',
    'Copper Cow Coffee': 'Ij6BNYohK2o',
    'The Original Stretchlace': 'FW4-8lfU-4o',
    "Float'N'Grill": 'wIb_J_nEOlY',
    'Frobert': "lPf4quRiB0A",
    'BeerMKR': 'J-oxoex0gys',
    'Grind Machine': 'FhCHwTzs5uM',
    'SwipenSnap': "8LiQZhlF_sQ",
    'Aura Bora': 'AjRoFuZ0MkE',
    'Brumachen': "OslUfkF1S8s",
    'TrophySmack': 'wuZGMr37ky8',
    'SneakErasers': None, # manually extracted
    'His & Her Bar': 'uVHQ8WLba68',
    'All33': 'ncJT0syzqx0',
    'Electra': 'yzytA958TlM',
    "See The Way I See": "NXOOMI2tju8&t=4s"

}
print("Total:", len(existing_videos.keys()))

Total: 119


# Task
Extract company names and video IDs from the descriptions in the "video_ids_&_description.json" file using the `gpt-4o-mini-2024-07-18` model, processing in batches of 10 with throttling, filtering for descriptions that sound like a startup pitch and removing compilations. The final output should be a dictionary mapping company names to video IDs.

## Load the json data

### Subtask:
Load the data from the "video_ids_&_description.json" file into a Python dictionary.


In [ ]:
import json

file_path = "video_ids_&_description.json"
with open(file_path, 'r') as f:
    video_data = json.load(f)

print(f"Loaded {len(video_data)} video descriptions.")

Loaded 299 video descriptions.


## Set up the api

### Subtask:
Install the necessary library for accessing the GPT API and configure it with your API key.


In [ ]:
%pip install openai

In [ ]:
from google.colab import userdata
import os

# Use the API key stored in Colab secrets
OPENAI_API_KEY = userdata.get("OPENAI_API_KEY")

## Define the processing function

### Subtask:
Create a function that takes a batch of video ID and description pairs, constructs prompts for the GPT model to identify startup pitches and extract company names, and returns the filtered results.


In [ ]:
import openai
import time
import json # Import the json library

# Initialize the OpenAI client with the API key
openai.api_key = OPENAI_API_KEY

def process_batch(batch_data):
    """
    Processes a batch of video ID and description pairs using GPT to identify startup pitches
    and extract company names.

    Args:
        batch_data: A dictionary where keys are video IDs and values are descriptions.

    Returns:
        A dictionary containing video IDs and extracted company names for startup pitches.
    """
    refined_results = {}
    for video_id, description in batch_data.items():
        # Skip entries with None description
        if description is None:
            continue

        # Exclude known compilation videos based on keywords in description
        if any(keyword in description.lower() for keyword in ["top 3", "biggest deals", "moments", "shark tank"]):
            continue

        prompt = f"""
        Analyze the following video description to determine if it describes a startup pitch on a show like Shark Tank.
        If it is a startup pitch, identify and extract the company name being pitched.
        Exclude descriptions that are compilations or highlight reels (e.g., "Top 10 moments", "Biggest deals", "Best of Shark Tank").
        Output the result in a JSON format with two keys:
        "is_pitch": boolean (true if it's a startup pitch, false otherwise)
        "videoId": string (the corresponding videoId)
        "company_name": string (the extracted company name, or null if not a pitch or no company name found)

        Video ID: {video_id}
        Description: {description}
        """

        try:
            response = openai.chat.completions.create(
                model="gpt-4o-mini-2024-07-18",
                response_format={"type": "json_object"},
                messages=[
                    {"role": "system", "content": "You are a helpful assistant that analyzes video descriptions."},
                    {"role": "user", "content": prompt}
                ]
            )
            result = json.loads(response.choices[0].message.content)

            if result.get("is_pitch") and result.get("company_name"):
                refined_results[video_id] = result["company_name"]
        except Exception as e:
            print(f"Error processing video ID {video_id}: {e}")
            # Implement a small delay before retrying or moving to the next item
            time.sleep(1) # Simple throttling

    return refined_results

print("Function process_batch defined.")

Function process_batch defined.


In [ ]:
import time

company_video_ids = {}
batch_size = 10
video_items = list(video_data.items())

for i in range(0, len(video_items), batch_size):
    batch_items = dict(video_items[i:i + batch_size])
    print(f"Processing batch {int(i/batch_size) + 1}...")
    batch_results = process_batch(batch_items)
    company_video_ids.update(batch_results)
    print(f"Finished batch {int(i/batch_size) + 1}. Waiting before next batch...")
    time.sleep(5) # Throttling

print(f"Finished processing all batches. Total companies found: {len(company_video_ids)}")

Processing batch 1...
Finished batch 1. Waiting before next batch...
Processing batch 2...
Finished batch 2. Waiting before next batch...
Processing batch 3...
Finished batch 3. Waiting before next batch...
Processing batch 4...
Finished batch 4. Waiting before next batch...
Processing batch 5...
Finished batch 5. Waiting before next batch...
Processing batch 6...
Finished batch 6. Waiting before next batch...
Processing batch 7...
Finished batch 7. Waiting before next batch...
Processing batch 8...
Finished batch 8. Waiting before next batch...
Processing batch 9...
Finished batch 9. Waiting before next batch...
Processing batch 10...
Finished batch 10. Waiting before next batch...
Processing batch 11...
Finished batch 11. Waiting before next batch...
Processing batch 12...
Finished batch 12. Waiting before next batch...
Processing batch 13...
Finished batch 13. Waiting before next batch...
Processing batch 14...
Finished batch 14. Waiting before next batch...
Processing batch 15...
F

In [ ]:
print(company_video_ids.items())

dict_items([('0qJ0lmx4HYA', 'Uprising Food'), ('_ALNAzjGEw0', 'Wisp'), ('kGMVz6zMQk4', 'Pavlok'), ('vhgDo-stGq8', 'Little Elf'), ('4XW5A-O_NP8', 'No Mo-Stache'), ('8XTjflYbm7k', 'Pronotbev'), ('YLLSXnyepqA', 'Coatchex'), ('et_zjQxl8Sk', 'Amber'), ('Qqs2S6R8xXA', 'bodywalking massage therapy'), ('sXkmrD_oTVA', 'Tenikle'), ('WT9FcPGPjtQ', 'Rolodoc'), ('WfolLSJ1zFo', 'Rick Smith Las Vegas Magic Show'), ('uKJQLxByVrg', 'BeSomebody'), ('rZCM3FzvOqQ', 'Pete & Pedro'), ('YL_Efp4H6z0', 'The Transformation Factory'), ('lh21LcjB1Nk', 'Copa Di Vino'), ('o3IVz_jFXyQ', 'KidsLuv'), ('Bt_igQZPEaI', 'Cat Amazing'), ('GX33DV-zR-I', 'Slate'), ('Frro-4hm7tY', 'The Scrubbie'), ('HvtN7Do9FRs', 'Pashion Footwear'), ('EA7hdVFvYJk', 'The Aqua Boxing Glove'), ('KwCVhWDXlx4', 'OverEZ Chicken Coops'), ('iAmoMRvbDtE', 'Storage Scholars'), ('0dViu8lAQN0', 'Neuro'), ('MdA1JIUTEHc', 'CUPBOP'), ('043cZY_eYpI', 'Sanaia Applesauce'), ('qubndMSCcQY', 'LoveSync'), ('lSP73qbIynY', 'HairFin'), ('a3IGj7YWXuI', 'Surfset'), (

In [ ]:
# Check for video IDs in company_video_ids that are already in existing_videos
duplicated_video_ids = [video_id for video_id in company_video_ids.keys() if video_id in existing_videos.values()]

if duplicated_video_ids:
    print("Duplicated video IDs found:")
    for video_id in duplicated_video_ids:
        print(f"- {video_id}")
else:
    print("No duplicated video IDs found.")

Duplicated video IDs found:
- sXkmrD_oTVA
- iAmoMRvbDtE
- NAe24u3wmyw
- UWH0pikR5Tw
- 0alIIxlxQ3Y
- UJRO3CveD4k
- tXYV-b4O1a8
- y4Dqgk5CqD4
- C0IoF9LH7Ao
- tKX3D1rFDRw
- rtcWMsw9HwA
- ciD1XwhJWXk
- Wy0z2ravpCQ
- FhCHwTzs5uM
- et9jM8fezNw
- Kqa7cGTqmb8
- OslUfkF1S8s
- 30-fgqbOwPU


In [ ]:
# Create a new dictionary with unique video IDs
deduplicated_company_video_ids = {
    video_id: company_name
    for video_id, company_name in company_video_ids.items()
    if video_id not in existing_videos.values() # Filter out video IDs that are already in existing_videos
}

print(f"Created a new dictionary with {len(deduplicated_company_video_ids)} unique entries.")
print(deduplicated_company_video_ids)

Created a new dictionary with 131 unique entries.
{'0qJ0lmx4HYA': 'Uprising Food', '_ALNAzjGEw0': 'Wisp', 'kGMVz6zMQk4': 'Pavlok', 'vhgDo-stGq8': 'Little Elf', '4XW5A-O_NP8': 'No Mo-Stache', '8XTjflYbm7k': 'Pronotbev', 'YLLSXnyepqA': 'Coatchex', 'et_zjQxl8Sk': 'Amber', 'Qqs2S6R8xXA': 'bodywalking massage therapy', 'WT9FcPGPjtQ': 'Rolodoc', 'WfolLSJ1zFo': 'Rick Smith Las Vegas Magic Show', 'uKJQLxByVrg': 'BeSomebody', 'rZCM3FzvOqQ': 'Pete & Pedro', 'YL_Efp4H6z0': 'The Transformation Factory', 'lh21LcjB1Nk': 'Copa Di Vino', 'o3IVz_jFXyQ': 'KidsLuv', 'Bt_igQZPEaI': 'Cat Amazing', 'GX33DV-zR-I': 'Slate', 'Frro-4hm7tY': 'The Scrubbie', 'HvtN7Do9FRs': 'Pashion Footwear', 'EA7hdVFvYJk': 'The Aqua Boxing Glove', 'KwCVhWDXlx4': 'OverEZ Chicken Coops', '0dViu8lAQN0': 'Neuro', 'MdA1JIUTEHc': 'CUPBOP', '043cZY_eYpI': 'Sanaia Applesauce', 'qubndMSCcQY': 'LoveSync', 'lSP73qbIynY': 'HairFin', 'a3IGj7YWXuI': 'Surfset', 'fuM1Vh7OgfY': 'Larq', '2Mlx_O0oxqM': 'Tog Samphel', 'w1l0_ll_HIU': 'Outer', 'BScmK

In [ ]:
# Check for company names in deduplicated_company_video_ids that are already in existing_videos
overlapping_names = [
    company_name for company_name in deduplicated_company_video_ids.values()
    if company_name in existing_videos.keys()
]

if overlapping_names:
    print("Overlapping company names found:")
    for name in overlapping_names:
        print(f"- {name}")
else:
    print("No overlapping company names found.")

Overlapping company names found:
- 54 Thrones
- VaBroom


In [ ]:
# Create a new dictionary excluding entries with overlapping company names
final_company_video_ids = {
    video_id: company_name
    for video_id, company_name in deduplicated_company_video_ids.items()
    if company_name not in existing_videos.keys()
}

print(f"Created a final dictionary with {len(final_company_video_ids)} entries after removing overlapping names.")
print(final_company_video_ids)

Created a final dictionary with 129 entries after removing overlapping names.
{'0qJ0lmx4HYA': 'Uprising Food', '_ALNAzjGEw0': 'Wisp', 'kGMVz6zMQk4': 'Pavlok', 'vhgDo-stGq8': 'Little Elf', '4XW5A-O_NP8': 'No Mo-Stache', '8XTjflYbm7k': 'Pronotbev', 'YLLSXnyepqA': 'Coatchex', 'et_zjQxl8Sk': 'Amber', 'Qqs2S6R8xXA': 'bodywalking massage therapy', 'WT9FcPGPjtQ': 'Rolodoc', 'WfolLSJ1zFo': 'Rick Smith Las Vegas Magic Show', 'uKJQLxByVrg': 'BeSomebody', 'rZCM3FzvOqQ': 'Pete & Pedro', 'YL_Efp4H6z0': 'The Transformation Factory', 'lh21LcjB1Nk': 'Copa Di Vino', 'o3IVz_jFXyQ': 'KidsLuv', 'Bt_igQZPEaI': 'Cat Amazing', 'GX33DV-zR-I': 'Slate', 'Frro-4hm7tY': 'The Scrubbie', 'HvtN7Do9FRs': 'Pashion Footwear', 'EA7hdVFvYJk': 'The Aqua Boxing Glove', 'KwCVhWDXlx4': 'OverEZ Chicken Coops', '0dViu8lAQN0': 'Neuro', 'MdA1JIUTEHc': 'CUPBOP', '043cZY_eYpI': 'Sanaia Applesauce', 'qubndMSCcQY': 'LoveSync', 'lSP73qbIynY': 'HairFin', 'a3IGj7YWXuI': 'Surfset', 'fuM1Vh7OgfY': 'Larq', '2Mlx_O0oxqM': 'Tog Samphel', 'w

In [ ]:
import json

# Output the final dictionary in JSON format
print(len(final_company_video_ids.keys()))
print(json.dumps(final_company_video_ids, indent=4))

129
{
    "0qJ0lmx4HYA": "Uprising Food",
    "_ALNAzjGEw0": "Wisp",
    "kGMVz6zMQk4": "Pavlok",
    "vhgDo-stGq8": "Little Elf",
    "4XW5A-O_NP8": "No Mo-Stache",
    "8XTjflYbm7k": "Pronotbev",
    "YLLSXnyepqA": "Coatchex",
    "et_zjQxl8Sk": "Amber",
    "Qqs2S6R8xXA": "bodywalking massage therapy",
    "WT9FcPGPjtQ": "Rolodoc",
    "WfolLSJ1zFo": "Rick Smith Las Vegas Magic Show",
    "uKJQLxByVrg": "BeSomebody",
    "rZCM3FzvOqQ": "Pete & Pedro",
    "YL_Efp4H6z0": "The Transformation Factory",
    "lh21LcjB1Nk": "Copa Di Vino",
    "o3IVz_jFXyQ": "KidsLuv",
    "Bt_igQZPEaI": "Cat Amazing",
    "GX33DV-zR-I": "Slate",
    "Frro-4hm7tY": "The Scrubbie",
    "HvtN7Do9FRs": "Pashion Footwear",
    "EA7hdVFvYJk": "The Aqua Boxing Glove",
    "KwCVhWDXlx4": "OverEZ Chicken Coops",
    "0dViu8lAQN0": "Neuro",
    "MdA1JIUTEHc": "CUPBOP",
    "043cZY_eYpI": "Sanaia Applesauce",
    "qubndMSCcQY": "LoveSync",
    "lSP73qbIynY": "HairFin",
    "a3IGj7YWXuI": "Surfset",
    "fuM1Vh7OgfY

In [ ]:
import json

output_file = "refined_company_video_ids.json"
with open(output_file, 'w') as f:
    json.dump(final_company_video_ids, f, indent=4)

print(f"Final results saved to {output_file}")

Final results saved to refined_company_video_ids.json


# Task
Analyze the length of the "Full_Pitch" field in the "existing_refinend_pitches.json" file, calculate the average length, and identify pitches that are significantly shorter than the average.

## Load the json data

### Subtask:
Load the data from the "existing_refinend_pitches.json" file into a Python dictionary or list of dictionaries.


In [ ]:
import json

file_path = "existing_refined_pitches.json"
with open(file_path, 'r') as f:
    pitches_data = json.load(f)

print(f"Loaded {len(pitches_data)} pitch entries.")

Loaded 120 pitch entries.


## Extract and measure pitch lengths

### Subtask:
Iterate through the loaded data and extract the "Full_Pitch" value for each entry. Measure the length of each extracted pitch string.


In [ ]:
pitch_lengths = []

for pitch_entry in pitches_data:
    full_pitch = pitch_entry.get("Full_Pitch")
    if isinstance(full_pitch, str):
        pitch_lengths.append(len(full_pitch))

print(f"Extracted lengths for {len(pitch_lengths)} pitches.")

Extracted lengths for 119 pitches.


## Analyze pitch lengths

### Subtask:
Calculate descriptive statistics for the pitch lengths (e.g., median, quartiles) to get a better understanding of the distribution and identify a suitable threshold for "significantly shorter" pitches, potentially considering the user's suggestion of around 1.2k characters.


In [ ]:
import numpy as np

median_pitch_length = np.median(pitch_lengths)
q1_pitch_length = np.percentile(pitch_lengths, 25)
q3_pitch_length = np.percentile(pitch_lengths, 75)

print(f"Median pitch length: {median_pitch_length}")
print(f"First quartile (25th percentile): {q1_pitch_length}")
print(f"Third quartile (75th percentile): {q3_pitch_length}")

Median pitch length: 1345.0
First quartile (25th percentile): 1144.5
Third quartile (75th percentile): 1689.5


## Identify pitches below the threshold

### Subtask:
Based on the calculated descriptive statistics, identify pitches that are significantly shorter than the median length (1345 characters) and might need further attention.


In [ ]:
significantly_shorter_pitches = []
exceptionally_longer_pitches = []

for pitch_entry in pitches_data:
    full_pitch = pitch_entry.get("Full_Pitch")
    if isinstance(full_pitch, str):
        if len(full_pitch) < q1_pitch_length:
            significantly_shorter_pitches.append({"Product": pitch_entry.get("Product"), "Length": len(full_pitch)})
        elif len(full_pitch) > q3_pitch_length:
             exceptionally_longer_pitches.append({"Product": pitch_entry.get("Product"), "Length": len(full_pitch)})

print(f"Found {len(significantly_shorter_pitches)} pitches significantly shorter than the first quartile ({q1_pitch_length:.2f} characters):")
for pitch in significantly_shorter_pitches:
    print(f"- {pitch['Product']}: {pitch['Length']} characters")

print(f"\nFound {len(exceptionally_longer_pitches)} pitches exceptionally longer than the third quartile ({q3_pitch_length:.2f} characters):")
for pitch in exceptionally_longer_pitches:
    print(f"- {pitch['Product']}: {pitch['Length']} characters")

Found 30 pitches significantly shorter than the first quartile (1144.50 characters):
- Dough: 1143 characters
- Dude Wipes: 808 characters
- Pink Picasso: 948 characters
- R girl whiskey: 566 characters
- Room Heaven: 591 characters
- Wendy's Gnome Shop: 1085 characters
- Banana Loca: 910 characters
- Liberate: 936 characters
- Headlights: 1113 characters
- ta3: 1043 characters
- ootbox: 1144 characters
- Booty Bag: 1031 characters
- Go Oats: 1011 characters
- Chilling Reel: 1040 characters
- Kent: 973 characters
- Oogie Bear: 1012 characters
- Wubble: 752 characters
- Posture Pins: 554 characters
- Hummingbird Mask: 801 characters
- Prime Fix: 577 characters
- Baked Goods with Edible Flowers: 1121 characters
- Dapper Boy: 974 characters
- The Fidget Game: 1059 characters
- Virgo Spurgo: 1068 characters
- The Magic 5: 708 characters
- Super Potty Trainer: 672 characters
- LIT Handlers: 756 characters
- N/A: 653 characters
- Aura Bora: 881 characters
- N/A: 3 characters

Found 30 pitche